docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,214 @@
+			 * Name: mpu_user_flash
+			 *
+			 * Description:
+			 *   Configure a region for user program flash
+			 *
+			 * Params:
+			 *  region : MPU region number
+			 *  base   : MPU region base address
+			 *  size   : MPU region size
+			 ****************************************************************************/
+			static inline void mpu_userflash(uint32_t region, uintptr_t base, size_t size);
+			```
+			```c
+			/****************************************************************************
+			 * Name: mpu_userintsram
+			 *
+			 * Description:
+			 *   Configure a region as user internal SRAM
+			 *
+			 * Params:
+			 *  region : MPU region number
+			 *  base   : MPU region base address
+			 *  size   : MPU region size
+			 ****************************************************************************/
+			static inline void mpu_userintsram(uint32_t region, uintptr_t base, size_t size);
+			```
+		- For repeated configuration of a MPU region, a two-step process is required using below functions:
+			1. In the first step, we obtain the mpu register values and store them in an array using below function:
+			```c
+			/****************************************************************************
+			 * Name: mpu_get_register_value
+			 *
+			 * Description:
+			 *   Configure the user application SRAM mpu settings into the tcb variables
+			 *
+			 * Params:
+			 *  regs     : pointer to array in which to store the configured values
+			 *             If regs is NULL, then configure the mpu registers directly
+			 *  region   : number of the region to be configured
+			 *  base     : start address for the region
+			 *  size     : size of the region in bytes
+			 *  readonly : true indicates a readonly region
+			 *  execute  : true indicates that the region has execute permission
+			 ****************************************************************************/
+			 
+			void mpu_get_register_value(uint32_t *regs, uint32_t region, uintptr_t base, size_t size, uint8_t readonly, uint8_t execute);
+			```
+			2. In the second step, the values in the array are used to configure the mpu. This second step happens everytime during a context switch.
+			```c
+			/****************************************************************************
+			 * Name: up_set_mpu_app_configuration
+			 *
+			 * Description:
+			 *   Configure MPU for app using register values stored in rtcb's mpu_regs[]
+			 *
+			 * Params:
+			 *  rtcb : Common part of Task Control Block
+			 ****************************************************************************/
+			static inline void up_set_mpu_app_configuration(struct tcb_s *rtcb);
+			```
+3. **Example : Adding new MPU region, MPU_REG_NUM_MY_REGION, with RO setting**
+	1. Add your MPU region in enum MPU_REG_NUM in common mpu.h file.
+	```c
+	enum MPU_REG_NUM {
+	 ....
+	 MPU_REG_NUM_MY_REGION,
+	 MPU_REG_NUM_MAX
+	};
+	```
+	2. To configure this MPU region as RO:
```

Comment:
You call only one function, mpu_get_register_value. It just combine the mpu register value. It does not set.
As you wrote two steps, you must call two functions (or one function which does all like mpu_userintsram). This example does not match with explanation.

Reply:
@sunghan-chang In description of `mpu_get_register_value(...)` above this, it is mentioned that if `regs` is passed as `NULL`, then mpu registers are configured directly.
In that case, no need for step 2.
But I will still update the example to 2 steps if you want

Reply:
No. Could you check that function? 
```c
	if (regs) {
		regs[0] = region;
		regs[1] = (base & MPU_RBAR_ADDR_MASK) | region;
		regs[2] = regval;
	} else {
#ifdef CONFIG_SUPPORT_COMMON_BINARY
		putreg32(region, MPU_RNR);
		putreg32((base & MPU_RBAR_ADDR_MASK) | region, MPU_RBAR);
		putreg32(regval, MPU_RASR);
#endif
	}
```
When CONFIG_SUPPORT_COMMON_BINARY is enabled, it's true. If you want to say like that, it should be working without any conditional. 
Additionally, if it is working like that, name and operation is not matched. We should renmae it.

Reply:
@kishore-sn It was merged with supporting common binary. But I think it should be modified.
1. That's the MPU API so that it should be working regardless of other feature.
2. It should be matched with name and operation.

Reply:
I agree for your comment where name and operation should match. We need to change source and modify the document accordingly? or we complete doc and create PR.

Reply:
Added new generic function `up_mpu_set_register(...)` which user can use to set mpu registers regardless of feature.

Reply:
Updated doc accordingly